include/picongpu/particles/ParticlesInit.kernel

```diff
@@ -64,6 +64,10 @@ namespace picongpu
         );
 
         float_X const densityRatioOfSpecies = traits::GetDensityRatio< T_Species >::type::getValue( );
+        PMACC_CASSERT_MSG(
+            Please_set_the_densityRatio_to_a_value_greater_than_0,
+            densityRatioOfSpecies > float_X( 0.0 )
```

Comment:
density ratio is not `constexpr` and can therefore not checked at compile time. We also have a PR #3109  where we started to introduce a check for `MassRatioValue` but we discarded it. We like to move more stuff into runtime (if possible).

Reply:
Removed from the commit